### PR TITLE
Add olm integration files for syndesis

### PR DESCRIPTION
**Pull request by lgarciaaco**
Based on our previous discussions I created this new PR, some considerations:

 - This is an operator for upstream, I have replaced Fuse Online references for Syndesis
 - It is using the [manifests format](https://github.com/operator-framework/operator-registry#manifest-format) recommended by [operator-courier](https://github.com/operator-framework/operator-courier)
 - All relevant files are now under `deploy` dir
 - The package version is 1.7.x while the operator version will be the latest within 1.7.x range

Observations:
 - I couldn't come up with that much for the description. Having a look at other operators, ppl have a sections describing the accepted values of the CR, but since we don't use the CR for reconciling (as far as I know), this section fell flat for us, unless we want to comment that our CR can be empty. Maybe we can describe the update procedure here (atm the operator is listed with install only capabilities)
 - The clusterPermission section is pretty much an abomination, but needed with the current version of the operator. The operator ServiceAccount passes down the edit ClusterRole to some of the syndesis service accounts, most likely syndesis-server. I will figure out what specifically is needed and narrow this down.

### CODE REVIEW

**Comment by lgarciaaco** [comment]
@heiko-braun @astefanutti @nicolaferraro we can start the discussion here one more time :-)

File: install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,933 @@
...
+metadata:
+  name: syndesisoperator.1.7.x
+  namespace: placeholder
+  annotations:
+    capabilities: Basic Install
+    categories: "Integration & Delivery"
+    certified: "false"
+    createdAt: 2019-05-08T16:12:00Z
+    containerImage: syndesis/syndesis-operator
+    support: Red Hat Fuse Online
```
Comment (zregvart) [inline install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml]:
```suggestion
    support: Syndesis
```

Not sure actually what value should this hold? Can't find any documentation in [the reference](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/building-your-csv.md) for the annotations used.

File: install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,933 @@
...
+    ### How to install
+    When the operator is installed (you have created a subscription and the operator is running in the selected namespace) create a new CR of Kind Syndesis (click the Create New button). The CR spec contains all defaults (see below).
+
+    ### CR Defaults
+    The CR definition is pretty simple and an empy CR will trigger a base installation
+  keywords: ['camel', 'integration', 'syndesis', 'fuse', 'online']
+  version: 1.7.x
+  maturity: alpha
+  maintainers:
+  - name: Red Hat Fuse Online
```
Comment (zregvart) [inline install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml]:
```suggestion
  - name: Syndesis team
```

File: install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,933 @@
...
+    email: fuse-online@redhat.com
```
Comment (zregvart) [inline install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml]:
```suggestion
    email: syndesis@googlegroups.com
```

File: install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,933 @@
...
+  provider:
+    name: Red Hat
```
Comment (zregvart) [inline install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml]:
```suggestion
    name: Syndesis team
```

File: install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,933 @@
...
+    mediatype: image/png
+  links:
+  - name: Syndesis
+    url: https://github.com/syndesisio/syndesis
```
Comment (zregvart) [inline install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml]:
```suggestion
    url: https://syndesis.io
```

File: install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,933 @@
...
+  - name: Syndesis Operator
```
Comment (zregvart) [inline install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml]: Not sure we need this, do we?

File: install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml (hunk 3 above)
Comment (lgarciaaco) [reply install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml]: not sure about what we want to reference here, @heiko-braun ?

File: install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml (hunk 2 above)
Comment (lgarciaaco) [reply install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml]: not sure about what we want to reference here, @heiko-braun?

Comment (heiko-braun) [reply install/operator/deploy/manifests/1.7.x/syndesisoperator.1.7.x.clusterserviceversion.yaml]: For upstream Zoran's suggestion seem to be right

**Pull request merged**